mlflow/server/handlers.py

```diff
@@ -1136,29 +1136,29 @@ def search_datasets_handler():
 
 @catch_mlflow_exception
 def gateway_proxy_handler():
-    gateway_uri = MLFLOW_GATEWAY_URI.get()
-    if not gateway_uri:
+    target_uri = MLFLOW_DEPLOYMENTS_TARGET.get()
+    if not target_uri:
         # Pretend an empty gateway service is running
-        return {"routes": []}
+        return {"endpoints": []}
 
     args = request.args if request.method == "GET" else request.json
 
     gateway_path = args.get("gateway_path")
     if not gateway_path:
         raise MlflowException(
-            message="GatewayProxy request must specify a gateway_path.",
+            message="Deployments proxy request must specify a gateway_path.",
             error_code=INVALID_PARAMETER_VALUE,
         )
     request_type = request.method
     json_data = args.get("json_data", None)
 
-    response = requests.request(request_type, f"{gateway_uri}/{gateway_path}", json=json_data)
+    response = requests.request(request_type, f"{target_uri}/{gateway_path}", json=json_data)
```

Comment:
do we need to handle the case where the value set in `MLFLOW_DEPLOYMENTS_TARGET` is not a uri? for example, "databricks" or "openai" (in the future).

also, rather than using python requests directly, should we just use the deploy client? 
```python
    from mlflow.deployments import get_deploy_client
    client = get_deploy_client() # this will use the value from get_deployment_target
    response = client.predict(endpoint=gateway_path, inputs=json_data)
```

Reply:
Good point. This function is used to list endpoints. The frontend only expects the OSS endpoint structure and doesn't work with databricks endpoints.

Reply:
@dbczumar Is this a blocker for 2.9?